src/lib/output/themes/default/assets/typedoc/Hierarchy.ts

```diff
@@ -0,0 +1,339 @@
+export interface HierarchyElement {
+    html: string;
+    text: string;
+    class: string;
+    path?: string;
+    kind?: number;
+    parents?: HierarchyElement[];
+    children?: HierarchyElement[];
+    depth?: number;
+}
+
+declare global {
+    interface Window {
+        // Base64 encoded data url, gzipped, JSON encoded HierarchyElement[]
+        hierarchyData?: string;
+    }
+}
+
+export function initHierarchy() {
+    const script = document.getElementById("tsd-hierarchy-script");
+    if (!script) return;
+
+    script.addEventListener("load", buildHierarchy);
+    buildHierarchy();
+}
+
+async function buildHierarchy() {
+    const container = document.getElementById("tsd-hierarchy-container");
+    if (!container || !window.hierarchyData) return;
+
+    const res = await fetch(window.hierarchyData);
+    const data = await res.arrayBuffer();
+    const json = new Blob([data])
+        .stream()
+        .pipeThrough(new DecompressionStream("gzip"));
+
+    const hierarchy = loadJson(await new Response(json).json());
+
+    let baseUrl = container.dataset.base!;
+
+    if (!baseUrl.endsWith("/")) baseUrl += "/";
+
+    const targetPath = container.dataset.targetPath;
+
+    const seeds = !targetPath
+        ? hierarchy.slice()
+        : hierarchy.filter((element) => {
+              if (element.path === targetPath) {
+                  element.class += " tsd-hierarchy-target";
+
+                  element.parents?.forEach(
+                      (parent) =>
+                          (parent.class += " tsd-hierarchy-close-relative"),
+                  );
+
+                  element.children?.forEach(
+                      (child) =>
+                          (child.class += " tsd-hierarchy-close-relative"),
+                  );
+
+                  return true;
+              }
+
+              if (
+                  !element.parents?.some(({ path }) => path === targetPath) &&
+                  !element.children?.some(({ path }) => path === targetPath)
+              ) {
+                  element.class += " tsd-hierarchy-distant-relative";
+
+                  return false;
+              }
+
+              return false;
+          });
+
+    const trees = getTrees(seeds, !targetPath);
+
+    if (!trees.length) return;
+
+    container
+        .querySelectorAll("ul.tsd-full-hierarchy")
+        ?.forEach((list) => list.remove());
+
+    trees.forEach((tree) => {
+        const list = buildList(
+            tree.filter((branch) => !branch.parents?.length),
+            baseUrl,
+            !targetPath,
+        )!;
+
+        list.classList.add("tsd-full-hierarchy");
+
+        container.append(list);
+    });
+
+    if (!targetPath && window.location.hash) {
```

Comment:
Currently the "view full" link includes a hash to the interface/class on the `hierarchy.html` page. To emulate that behavior for content that was loaded dynamically this work around is required, but if I rework the PR to have `hierarchy.html` be static as seems to be the desire in a comment below it can easily be removed.

Reply:
Removed